src/game_inventory.cpp

```diff
@@ -154,24 +154,100 @@ item *game::inv_map_for_liquid( const item &liquid, const std::string &title, in
+            append_cell( [ this ]( const item_location & loc ) -> std::string {
+                const auto odds = get_odds( loc );
+
+                if( odds.first >= 100 ) {
+                    return string_format( "<color_ltgreen>%s</color>", _( "always" ) );
+                }
+
+                return string_format( "<color_ltgreen>%d%%</color>", odds.first );
+            }, _( "SUCCESS" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                const auto odds = get_odds( loc );
+                return odds.second > 0 ? string_format( "<color_red>%d%%</color>", odds.second ) : std::string();
+            }, _( "DAMAGE" ) );
+        }
+
+        bool is_shown( const item_location &loc ) const override {
+            return loc->is_gun() && !loc->is_gunmod();
+        }
+
+        std::string get_denial( const item_location &loc ) const override {
+            std::string incompatability;
+
+            if( !loc->gunmod_compatible( gunmod, &incompatability ) ) {
+                return incompatability;
+            }
+
+            if( get_odds( loc ).first <= 0 ) {
+                return _( "is too difficult for you to modify" );
+            }
+
+            const auto &min_skills = gunmod.type->min_skills;
+            const auto &gun_skill = loc->gun_skill();
+            const auto iter = min_skills.find( "weapon" );
+
+            if( iter != min_skills.end() && p.get_skill_level( gun_skill ) < iter->second ) {
+                //~ %1$s - name of the skill, %2$d - required value of the skill
+                return string_format( _( "requires %1$s skill %2$d to be modified" ), gun_skill.c_str(),
+                                      iter->second );
+            }
+
+            return std::string();
+        }
+
+        bool sort_compare( const item_location &lhs, const item_location &rhs ) const override {
+            const auto a = get_odds( lhs );
+            const auto b = get_odds( rhs );
+
+            if( a.first > b.first || ( a.first == b.first && a.second < b.second ) ) {
+                return true;
+            }
+
+            return inventory_selector_preset::sort_compare( lhs, rhs );
+        }
+
+    protected:
+        std::pair<int, int> get_odds( const item_location &gun ) const {
+            return p.gunmod_installation_odds( *gun, gunmod );
+        }
+
+    private:
+        const player &p;
+        const item &gunmod;
+};
+
+item_location game::inv_for_gunmod( const item &gunmod, const std::string &title )
+{
+    return inv_internal( u, gunmod_inventory_preset( u, gunmod ),
+                         title, -1, _( "You don't have any guns to modify." ) );
 }
 
 std::list<std::pair<int, int>> game::multidrop()
 {
     u.inv.restack( &u );
     u.inv.sort();
 
-    inventory_filter_preset preset(
-    [ this ]( const item_location & location ) -> bool {
+    const inventory_filter_preset preset( [ this ]( const item_location & location ) {
```

Comment:
What do you mean? I don't see any, except the space after `&` which is a glitch of astyle.